### PR TITLE
v2 endpoints to list OpenStack resources

### LINKED ISSUE #6206: v2 list OpenStack resources
GET /api/v2/projects/{project_id}/clusters/{cluster_id}/providers/openstack/availabilityzones
GET /api/v2/projects/{project_id}/clusters/{cluster_id}/providers/openstack/subnets
GET /api/v2/projects/{project_id}/clusters/{cluster_id}/providers/openstack/securitygroups
GET /api/v2/projects/{project_id}/clusters/{cluster_id}/providers/openstack/networks
GET /api/v2/projects/{project_id}/clusters/{cluster_id}/providers/openstack/tenants
GET /api/v2/projects/{project_id}/clusters/{cluster_id}/providers/openstack/sizes

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: v2 endpoints to list OpenStack resources

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #6206

```release-note
v2 endpoints to list OpenStack resources:
GET /api/v2/projects/{project_id}/clusters/{cluster_id}/providers/openstack/availabilityzones
GET /api/v2/projects/{project_id}/clusters/{cluster_id}/providers/openstack/subnets
GET /api/v2/projects/{project_id}/clusters/{cluster_id}/providers/openstack/securitygroups
GET /api/v2/projects/{project_id}/clusters/{cluster_id}/providers/openstack/networks
GET /api/v2/projects/{project_id}/clusters/{cluster_id}/providers/openstack/tenants
GET /api/v2/projects/{project_id}/clusters/{cluster_id}/providers/openstack/sizes
```

### CODE REVIEW

File: pkg/handler/common/provider/openstack.go [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+	userInfo, err := userInfoGetter(ctx, projectID)
+	if err != nil {
+		return nil, common.KubernetesErrorToHTTPError(err)
+	}
+
+	secretKeySelector := provider.SecretKeySelectorValueFuncFactory(ctx, assertedClusterProvider.GetSeedClusterAdminRuntimeClient())
+	creds, err := openstack.GetCredentialsForCluster(cluster.Spec.Cloud, secretKeySelector)
+	if err != nil {
+		return nil, err
+	}
```
Comment (lsviben) [inline pkg/handler/common/provider/openstack.go]: this is repeated multiple times, we could extract it to getCredentials or getCreds. Maybe without the userInfoGetter check

**Review by lsviben — COMMENTED** [review]
Looks ok, one small comment

File: pkg/handler/common/provider/openstack.go (hunk 1 above)
Comment (zreigz) [reply pkg/handler/common/provider/openstack.go]: done

**Pull request merged**